### PR TITLE
bug 1387100 - improve webapp in dev environment

### PR DESCRIPTION
**Pull request by willkg**
This adds lookup table data population and documented run lines that add release
data to the db. These make it possible to go to
http://localhost:8000/home/product/Firefox and have it mostly work.

Note that this only pulls release data for Firefox. Each additional product
takes another like 10 minutes to run. The thinking is that we're primarily
working on Firefox crashes, so if someone needs release data for something else,
they can adjust the scripts accordingly. If we need to make that more flexible,
we can make that change later.

### CODE REVIEW

**Comment by willkg** [comment]
The goal here is that this gets you to a mostly functional Firefox homepage:

1. `make dockerbuild`
2. `make dockersetup`
3. `docker-compose run processor docker/run_update_release_data.sh`
4. `docker-compose up webapp`

Then you browse to ``http://localhost:8000`` and get a Firefox home page that has active versions, but doesn't have any ADI or crash data.

**Comment by peterbe** [comment]
One that we're going to need to do is to get fancy with the `collectstatic` stuff. It's a bore to have to wait for since when you run with DEBUG=False you don't need static assets collected first. In fact, with DEBUG=True the collect static files aren't being used. 

Devs need to be able to test running the webapp with DEBUG=False and javascript files minified. But generally this is only needed in CI after the tests pass and artifacts are built.

**Comment by willkg** [comment]
I agree--that's definitely worth spinning out into a separate bug because it's tricky.

File: socorro/external/postgresql/staticdata.py [hunk 1]
```diff
@@ -189,8 +212,79 @@ class ReportPartitionInfo(BaseTable):
...
+        ['Firefox', '1', '5.0', 'firefox', '23.0'],
+        ['SeaMonkey', '6', '2.3', 'seamonkey', '999.0'],
+        ['FennecAndroid', '4', '5.0', '**SPECIAL**', '999.0'],
+    ]
+
+
+class ProductBuildTypes(BaseTable):
+    table = 'product_build_types'
+    columns = [
+        'product_name', 'build_type'
```
Comment (peterbe) [inline socorro/external/postgresql/staticdata.py]: Add `, 'throttle'` here and ...

File: socorro/external/postgresql/staticdata.py [hunk 2]
```diff
@@ -189,8 +212,79 @@ class ReportPartitionInfo(BaseTable):
...
+    ]
+    rows = [
+        ['Firefox', 'release'],
```
Comment (peterbe) [inline socorro/external/postgresql/staticdata.py]: Make this `['Firefox', 'release', 0.1],` and for all other 2-item lists make it `1.0` instead of `0.1`. E.g. `['Firefox', 'beta', 1.0]`.

File: socorro/external/postgresql/staticdata.py [hunk 3]
```diff
@@ -100,14 +100,37 @@ class OSVersions(BaseTable):
...
+    rows = [
+        ['mozilla-central'],
+        ['mozilla-1.9.2'],
+        ['comm-central'],
+        ['comm-1.9.2'],
+        ['comm-central-trunk'],
+        ['mozilla-central-android'],
+        ['mozilla-release'],
+        ['mozilla-beta'],
+        ['mozilla-aurora'],
```
Comment (willkg) [inline socorro/external/postgresql/staticdata.py]: I want this to be -prod like. I'm not wildly excited about removing things here that are in -prod. When we remove the aurora stuff in bug #1385285 and assuming that covers references in the db, too, I'd be game for removing it in staticdata. Until then, I want to keep it.

File: socorro/external/postgresql/staticdata.py (hunk 1 above)
Comment (willkg) [inline socorro/external/postgresql/staticdata.py]: Huh... I'm not sure how I forgot that. I'll add those now.

File: docker/run_update_release_data.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
```
Comment (willkg) [inline docker/run_update_release_data.sh]: Yes! I'll add that now.

File: docs/gettingstarted.rst [hunk 5]
```diff
@@ -41,24 +41,66 @@ Quickstart
...
+   which can take a long time, maybe hours.
+
+
+General architecture
+====================
+
+.. image:: block-diagram.png
+
+Arrows direction represents the flow of interesting information (crashes,
+authentication assertions, cached values), not trivia like acks.
```
Comment (willkg) [inline docs/gettingstarted.rst]: Yeah. I want to switch this to dot and graphviz, but haven't, yet. I'm using that with Antenna and it's been good:

http://antenna.readthedocs.io/en/latest/architecture.html#high-level-architecture

It has the advantage of being text-based and easier to edit in the docs:

https://github.com/mozilla-services/antenna/blob/master/docs/architecture.rst#high-level-architecture

I think that's outside the scope of this pass, though.

**Comment by willkg** [comment]
> I think the homepage is failing because there is absolutely no crashes in ES. Should we bother updating the JS code to not choke if there is exactly 0 crashes?

It's definitely from missing data. When we talked yesterday, you said you'd work on building a list of all the tables the product home page needs to work completely. I know this PR covers some/most of them. I know it's missing ADI data (I'm working on that in a separate PR). I know it's missing crash data. Beyond that, I have no idea what it needs and the code is hard to follow.

That's why this PR limits the scope to "get the things required so we don't get an HTTP 404 on the product home page anymore".

Towards updating the JS code to not choke, if that's not too hard to do, I think it's worth doing because it improves the ergonomics of the dev environment.

> How can I help write a thing that populates the ADI from prod? Thing is what's in the public ADIs is the materialized view. To populate locally we'd need to first fill in raw_adi then execute the update_adu.sql stored procedure so it combines the raw_adi table with product_versions normalized.

I'm working on a script now that pulls ADI data from the API and populates postgres. I'll let you know if I hit problems.

> I'm getting this traceback ...

I don't get any tracebacks when running the webapp. I'm guessing either you have a `.env` file that's different or maybe you've got additional bits in your `socorro/` directory that aren't part of the repo maybe from running Socorro on the host or something along those lines.

> SuperSearchFields isn't working. And that's a whole lot of other work but I'm not interested in solving this the old way (populating ES with the supersearch fields) until we can start using the super_search_fields.json file that is currently "pending" until we upgrade to ES 5.

I'd like to get supersearch working, but I agree we should wait until `super_search_fields.json` lands. Depending on how long that takes, I might take that work and land it before ES 5. I'll figure that out later, though.

I think that covers all the outstanding things in this issue.

**Comment by willkg** [comment]
Travis is fine, so I'm going to land this.

**Comment by peterbe** [comment]
Yeah. This was the PR I was waiting for to see what the home page needs. 
I think it's simpler and harder than we imagined. There's nothing else needed in lookup tables needed. However, it won't work without something out of supersearch and without any adi it won't error but it'll simply not show the line in the graph. 

Let's worry about the JS error (if the supersearch result is empty) until you've made the other progress.
